### PR TITLE
Script for automatic doc testing

### PR DESCRIPTION
**Pull request by ajdapretnar**
##### Issue

When new documentation is added, widget.json needs to be updated, too.

##### Description of changes
Add script to automatically check documentation updates.

##### Includes
- [X] Code changes
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by markotoplak** [comment]
@ajdapretnar, rebase this to master, please.

**Comment by ajdapretnar** [comment]
@markotoplak Oh gawd, now Travis is complaining it doesn't have WebEngine... Should this be put in requirements-doc.txt or what?

**Comment by robertcv** [comment]
Is it necessary to build documentation with miniconda?

File: .travis.yml [hunk 1]
```diff
@@ -2,6 +2,11 @@ language: python
 
 dist: xenial
 
+addons:
```
Comment (robertcv) [inline .travis.yml]: This is now already on master, please remove.

File: .travis/build_doc.sh [hunk 2]
```diff
@@ -2,5 +2,29 @@
 
 set -o errexit
 
+# PyQt >= 5.12 distributes WebEngine separately
+pip install pyqtwebengine
```
Comment (robertcv) [inline .travis/build_doc.sh]: This is not needed. Just remove the restriction on pyqt version in [conda_environment.yml](https://github.com/biolab/orange3-text/blob/629d9570ecec93b0e5d2a32c19959a9bd91b0343/conda_environment.yml#L15). The import qtwebengine error resolves itself.

File: .travis/build_doc.sh [hunk 3]
```diff
@@ -2,5 +2,29 @@
...
+
 # -W treats warnings as errors to make sure all went ok
 sphinx-build -W doc doc/_build
+
+XVFBARGS="-screen 0 1280x1024x24"
+
+# check if the widget catalog in the repository (for orange-hugo is up to date)
+cd $TRAVIS_BUILD_DIR/doc/
+wget_command="wget -N https://raw.githubusercontent.com/biolab/orange-hugo/master/scripts/create_widget_catalog.py"
+run_command="python create_widget_catalog.py --categories "Text\ Mining" --doc ."
```
Comment (robertcv) [inline .travis/build_doc.sh]: Use `\"Text Mining\"`

File: .travis/build_doc.sh [hunk 4]
```diff
@@ -2,5 +2,29 @@
...
+eval "$wget_command"
+eval "catchsegv xvfb-run -a -s "\"$XVFBARGS"\" $run_command"
+diff=$(git diff -- widgets.json)
+echo $diff
```
Comment (robertcv) [inline .travis/build_doc.sh]: Don't display the difference twice.

**Pull request merged**